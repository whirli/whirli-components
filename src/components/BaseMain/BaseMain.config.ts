import { PropType } from 'vue';
import * as PropHelpers from '../../helpers/props';
import { ComponentStyles, ComponentProps } from '../../@types/components';

// Local config
import config from '@whirli/BaseMain/BaseMain.config';

// Package config
import { PropKeys, PropValues } from './BaseMain.constants';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
