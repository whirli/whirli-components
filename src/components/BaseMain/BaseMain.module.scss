@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles
.main {}
// breakpoint styles
$packageBreakpointStyles: ();