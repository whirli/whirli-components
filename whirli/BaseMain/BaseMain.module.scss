@import '@whirli-components/src/components/BaseMain/BaseMain.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));