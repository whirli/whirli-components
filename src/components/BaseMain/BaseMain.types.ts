import { PropKeys, PropValues } from './BaseMain.constants';
