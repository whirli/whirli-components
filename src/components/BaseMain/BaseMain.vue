<template>
  <main id="main" :class="classes">
    <slot />
  </main>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseMain/BaseMain.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseMain.config';

const ComponentStyles = ConfigStyles;

// @ts-ignore
const props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes = [styles.main, ...makeClasses(ComponentStyles, ConfigProps, props, styles)];
</script>
