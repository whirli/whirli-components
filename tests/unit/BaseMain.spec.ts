import { shallowMount } from '@vue/test-utils';
import BaseMain from '@/components/BaseMain/BaseMain.vue';

let wrapper: any;
const text = 'Hello World';

describe('BaseMain.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseMain, {
      slots: {
        default: text,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.text()).toMatch(text);
  });
});
