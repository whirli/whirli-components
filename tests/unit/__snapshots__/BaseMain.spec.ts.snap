// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseMain.vue - Single classes renders the component 1`] = `
<main
  class=""
  id="main"
>
  
  Hello World
  
</main>
`;
